config.c

```diff
@@ -377,65 +404,73 @@ bool apply_superset_style(
 	return true;
 }
 
-static bool apply_config_option(struct mako_config *config, const char *name,
-		const char *value) {
+static bool apply_surface_option(struct mako_surface_config *surface,
+		const char *name, const char *value) {
 	if (strcmp(name, "max-visible") == 0) {
-		return parse_int(value, &config->max_visible);
+		return parse_int(value, &surface->max_visible);
 	} else if (strcmp(name, "max-history") == 0) {
-		return parse_int(value, &config->max_history);
+		return parse_int(value, &surface->max_visible);
```

Comment:
Copypasta accident.

Reply:
More generally, do we want history to be per-surface? I don't think this code currently works as expected because we're still tracking history globally. I think it makes more sense to keep it global and provide mechanisms for excluding specific notifications (volume indicators, etc) from history entirely.

Reply:
TBH. I mostly rebased onto that syntactically.

Probably has the history on `mako_state` while the max_history is currently on the surface, because that's how git auto-merged.

I'll have to get a grasp of history in general. Would it make sense to have the exclusion on the surface, or per notification?

Reply:
Ahh, I just saw that there's a style option for it.